app/delivery/send_to_providers.py

```diff
@@ -336,16 +339,55 @@ def update_notification_to_sending(notification, provider):
     dao_update_notification(notification)
 
 
-def provider_to_use(notification_type, notification_id, international=False, sender=None, template_id=None):
-    # Temporary redirect setup for template IDs that are meant for the short code usage.
-    if notification_type == SMS_TYPE and template_id is not None and str(template_id) in Config.AWS_PINPOINT_SC_TEMPLATE_IDS:
-        return clients.get_client_by_name_and_type("pinpoint", SMS_TYPE)
+def provider_to_use(
+    notification_type: str,
+    notification_id: UUID,
+    to: Union[str, None] = None,
+    international: bool = False,
+    sender: Union[str, None] = None,
+) -> Any:
+    """
+    Get the provider to use for sending the notification.
+    SMS that are being sent with a dedicated number or to a US number should not use Pinpoint.
```

Comment:
Thanks for the documentation and typing annotations.

Reply:
I  figured I had added enough new logic to this function that some documentation was useful 😬